src/NetscriptFunctions/CodingContract.ts

```diff
@@ -65,8 +73,26 @@ export function NetscriptCodingContract(): InternalAPI<ICodingContract> {
       const filename = helpers.string(ctx, "filename", _filename);
       const hostname = _hostname ? helpers.string(ctx, "hostname", _hostname) : ctx.workerScript.hostname;
       const contract = getCodingContract(ctx, hostname, filename);
+
       return structuredClone(contract.getData());
     },
+    getContract: (ctx) => (_filename, _hostname?) => {
+      const filename = helpers.string(ctx, "filename", _filename);
+      const hostname = _hostname ? helpers.string(ctx, "hostname", _hostname) : ctx.workerScript.hostname;
+      const server = helpers.getServer(ctx, hostname);
+      const contract = getCodingContract(ctx, hostname, filename);
+      // asserting type here is required, since it is not feasible to properly type getData
+      return {
+        type: contract.type,
+        data: contract.getData(),
+        attempt: (answer: unknown) => {
```

Comment:
`attempt` is going to flag the static-ram detector. Not entirely sure what to do about that. Maybe `makeAttempt`?

Reply:
I thought of another possible answer here. Instead of calling `helpers.checkEnvFlags`, you could literally call through `ctx` to the netscript attempt method, thus invoking its dynamic ram cost. This would be a little strange, but at the same time make a certain kind of sense, and it would be compatible with how the static RAM checker works (the name would stay the same).
In this case, you wouldn't want to charge more for `getContract`. Also, you'd want to apply the same logic for `numTriesRemaining` (rename to `getNumTriesRemaining()`).

Reply:
I'm not quite sure that would be a good idea? I like that it's kinda an alias to the same function.
Relying on the static RAM checker to catch that call is a quite the gamble IMO.

My current plan was to rename the `attempt` method to `submit`. The RAM remarks in the JSDoc would simply state what it costs to use that function. Any methods on the object are included in that number. When the methods themselves invoke the static RAM cost, it could be a bit unintuitive to some players, because they think the static RAM from the `getContract` function includes everything.